packages/p2p-media-loader-hlsjs/src/services/fragment-loader.ts

```diff
@@ -34,32 +61,198 @@ export class FragmentLoaderBase implements Loader<FragmentLoaderContext> {
     this.context = context;
     this.config = config;
     this.callbacks = callbacks;
-    this.defaultLoader.load(context, config, {
-      ...callbacks,
-      onSuccess: (response, stats, context, networkDetails) => {
-        const { rangeStart: start, rangeEnd: end } = context;
-        const segmentId = Segment.getSegmentLocalId(context.url, {
-          start,
-          end: end !== undefined ? end - 1 : undefined,
-        });
-        const playlist = this.segmentManager.getPlaylistBySegmentId(segmentId);
-        this.debug(
-          "downloaded segment from playlist\n",
-          `playlist v: ${playlist?.index}\n`,
-          `segment: `,
-          playlist?.segments.get(segmentId)?.index
-        );
-
-        return callbacks.onSuccess(response, stats, context, networkDetails);
+    const stats = this.stats;
+
+    console.log(context.url);
+    const playlist = this.identifyPlaylist(context);
+    if (!playlist) {
+      this.defaultLoader = this.createDefaultLoader();
+      this.stats = this.defaultLoader.stats;
```

Comment:
Could you please check? Probably it should be different direction:
```suggestion
      this.defaultLoader.stats = this.stats;
```